mips/malta.c

```diff
@@ -0,0 +1,70 @@
+#include <mips/malta.h>
+#include <malloc.h>
+#include <stdc.h>
+
+void clear_bss() {
+  extern unsigned int __bss[];
+  extern unsigned int __ebss[];
+  bzero(__bss, __ebss - __bss);
+}
+
+#define ISSPACE(a) ((a) == ' ' || ((a) >= '\t' && (a) <= '\r'))
+
+/*
+ * For some reason arguments passed to the kernel are stored in one string -
+ * it means that argc is always equals 2 and argv[1] string contains all passed
+ * parameters. Because we want each parameter in separate argv entry
+ * (as it should be) we need to fix it. This function parse that argv[1]
+ * string, split it to separate arguments, creates new argv table with proper
+ * entries and corrects argc.
+ *
+ * Example:
+ *
+ *   before:
+ *     argc=2;
+ *     argv={"<program name>", "arg1 arg2=val   arg3=foobar  "};
+ *
+ *   fixing:
+ *     fix_argv(&argc, &argv);
+ *
+ *   after:
+ *     argc=4;
+ *     argv={"<program name>", "arg1", "arg2=val", "arg3=foobar"};
+ */
+
+void fix_argv(int *p_argc, char ***p_argv) {
+  if (*p_argc < 2)
+    return;
+
+  char *cmd = (*p_argv)[1];
+  char *s = cmd, *p = cmd;
+  int argc;
+  char **argv;
+
+  // argv[1] string is modified - all whitespaces are overwritten 
```

Comment:
Don't focus too much to that argv fix - we have much more problems to solve in this project. I know that my solution is not great but for now it should be enough. Just do not waste time trying to fix problems that has not even appeared yet and are unlikely to happen in near future. If they appear I will improve this function, but for now just keep it as it is.